diagnostics/tropical_pacific_sea_level/tropical_pacific_sea_level.py

```diff
@@ -0,0 +1,671 @@
+
+
+        # calculate regional mean
+        regionalavg_list['%s_%i_%i_%i_%i_season'%(var,lon_range[0],lon_range[1],lat_range[0],lat_range[1])]\
+          = ((season_mlist[model][var]*ds_mask*da_area).sum(dim=[xname,yname])/(ds_mask*da_area).sum(dim=[xname,yname])).compute()
+
+        regionalavg_list['%s_%i_%i_%i_%i_mean'%(var,lon_range[0],lon_range[1],lat_range[0],lat_range[1])]\
+          = ((mean_mlist[model][var]*ds_mask*da_area).sum(dim=[xname,yname])/(ds_mask*da_area).sum(dim=[xname,yname])).compute()
+
+        regionalavg_list['%s_%i_%i_%i_%i_linear'%(var,lon_range[0],lon_range[1],lat_range[0],lat_range[1])]\
+          = ((linear_mlist[model][var]*ds_mask*da_area).sum(dim=[xname,yname])/(ds_mask*da_area).sum(dim=[xname,yname])).compute()
+
+    regionalavg_mlist[model] = regionalavg_list
+
+#####################
+# OBS
+#####################
+obs_regionalavg_mlist = {}
+for nobs,obs in enumerate(Obs_name):
+    obs_regionalavg_list = {}
+    if obs in ['CMEMS']:
+        var = 'adt'
+        obs_xname = 'lon'
+        obs_yname = 'lat'
+
+    elif obs in ['WASwind']:
+        var = 'curl_tau'
+        obs_xname = 'lon'
+        obs_yname = 'lat'
+
+
+    da_area = sa.da_area(obs_mean_mlist[obs][var], lonname='lon', latname='lat',
+                                 xname=obs_xname, yname=obs_yname, model=None)
+
+    # crop region
+    ds_obs_mask = obs_mean_mlist[obs][var].where(
+                  (obs_mean_mlist[obs][var].lon>=np.min(lon_range_mod))&
+                  (obs_mean_mlist[obs][var].lon<=np.max(lon_range_mod))&
+                  (obs_mean_mlist[obs][var].lat>=np.min(lat_range))&
+                  (obs_mean_mlist[obs][var].lat<=np.max(lat_range))
+                  ,drop=True).compute()
+    ds_obs_mask = ds_obs_mask/ds_obs_mask
+
+    # calculate regional mean
+    obs_regionalavg_list['%s_%i_%i_%i_%i_season'%(var,lon_range[0],lon_range[1],lat_range[0],lat_range[1])]\
+      = ((obs_season_mlist[obs][var]*ds_obs_mask*da_area).sum(dim=[obs_xname,obs_yname])/ \
+         (ds_obs_mask*da_area).sum(dim=[obs_xname,obs_yname])).compute()
+
+    obs_regionalavg_list['%s_%i_%i_%i_%i_mean'%(var,lon_range[0],lon_range[1],lat_range[0],lat_range[1])]\
+      = ((obs_mean_mlist[obs][var]*ds_obs_mask*da_area).sum(dim=[obs_xname,obs_yname])/ \
+         (ds_obs_mask*da_area).sum(dim=[obs_xname,obs_yname])).compute()
+
+    obs_regionalavg_list['%s_%i_%i_%i_%i_linear'%(var,lon_range[0],lon_range[1],lat_range[0],lat_range[1])]\
+      = ((obs_linear_mlist[obs][var]*ds_obs_mask*da_area).sum(dim=[obs_xname,obs_yname])/ \
+         (ds_obs_mask*da_area).sum(dim=[obs_xname,obs_yname])).compute()
+
+    obs_regionalavg_mlist[obs] = obs_regionalavg_list
+
+
+#### plotting
+fig = plt.figure(1)
+
+
+#######
+# mean
+#######
+ax1 = fig.add_axes([0,0,1,1])
+obscolor = 'k'
+
+all_wsc = []
+all_ssh = []
+
+wsc = obs_regionalavg_mlist['WASwind']['%s_%i_%i_%i_%i_mean'%('curl_tau',lon_range[0],lon_range[1],lat_range[0],lat_range[1])]
```

Comment:
Consider using f-strings to make this more readable:  https://docs.python.org/3/tutorial/inputoutput.html

Reply:
Updated to F-strings for readability.